static/app/views/performance/browser/webVitals/utils/useTransactionSamplesWebVitalsQuery.tsx

```diff
@@ -0,0 +1,124 @@
+import {useDiscoverQuery} from 'sentry/utils/discover/discoverQuery';
+import EventView from 'sentry/utils/discover/eventView';
+import {useLocation} from 'sentry/utils/useLocation';
+import useOrganization from 'sentry/utils/useOrganization';
+import usePageFilters from 'sentry/utils/usePageFilters';
+import {calculatePerformanceScore} from 'sentry/views/performance/browser/webVitals/utils/calculatePerformanceScore';
+import {
+  TransactionSampleRowWithScore,
+  WebVitals,
+} from 'sentry/views/performance/browser/webVitals/utils/types';
+
+type Props = {
+  transaction: string;
+  limit?: number;
+  orderBy?: WebVitals | null;
+  query?: string;
+};
+
+export const useTransactionSamplesWebVitalsQuery = ({
+  orderBy,
+  limit,
+  transaction,
+  query,
+}: Props) => {
+  const organization = useOrganization();
+  const pageFilters = usePageFilters();
+  const location = useLocation();
+
+  const eventView = EventView.fromNewQueryWithPageFilters(
+    {
+      fields: [
+        'id',
+        'user.display',
+        'transaction',
+        'transaction.op',
+        'measurements.lcp',
+        'measurements.fcp',
+        'measurements.cls',
+        'measurements.ttfb',
+        'measurements.fid',
+        'transaction.duration',
+        'replayId',
+        'timestamp',
+      ],
+      name: 'Web Vitals',
+      query: `transaction.op:pageload transaction:"${transaction}" ${query ? query : ''}`,
+      orderby: mapWebVitalToOrderBy(orderBy),
+      version: 2,
+    },
+    pageFilters.selection
+  );
+
+  const {data, isLoading, ...rest} = useDiscoverQuery({
+    eventView,
+    limit: limit ?? 50,
+    location,
+    orgSlug: organization.slug,
+    options: {
+      enabled: pageFilters.isReady,
+      refetchOnWindowFocus: false,
+    },
+  });
+
+  const tableData: TransactionSampleRowWithScore[] =
+    !isLoading && data?.data.length
+      ? data.data
+          .map(row => ({
+            id: row.id?.toString(),
+            'user.display': row['user.display']?.toString(),
+            transaction: row.transaction?.toString(),
+            'transaction.op': row['transaction.op']?.toString(),
+            'measurements.lcp': row['measurements.lcp'] as number,
```

Comment:
m: should we be doing this cast? Given the web vitals are not always defined anyway (and sometimes we are explicitly even filtering via `!has:measurements.lcp`)

Reply:
Removed the cast and updated the code to explicitly check for null